airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/BufferingStrategy.java

```diff
@@ -40,12 +45,4 @@ public interface BufferingStrategy extends AutoCloseable {
    */
   void clear() throws Exception;
 
-  /**
-   * When all buffers are being flushed, we can signal some parent function of this event for further
-   * processing.
-   *
-   * THis install such a hook to be triggered when that happens.
-   */
-  void registerFlushAllEventHook(VoidCallable onFlushAllEventHook);
```

Comment:
yes, the "flush all event" is indeed a good smell to be looked at, thank you for investigating this.

The buffering strategy refactor was already a pretty large PR on handling AirbyteMessage differently, handling the state messages in a good manner was not necessarily the primary priority there... So the hook was introduced to avoid diverging too much from the previous behavior of destinations on that matter.